marshmallow_dataclass/__init__.py

```diff
@@ -88,31 +92,48 @@ def dataclass(_cls: type = None, *,
     Point(x=0.0, y=0.0)
     """
     dc = dataclasses.dataclass(_cls, repr=repr, eq=eq, order=order, unsafe_hash=unsafe_hash, frozen=frozen)
-    return add_schema(dc) if _cls else lambda cls: add_schema(dc(cls))
+    return add_schema(dc, base_schema) if _cls else lambda cls: add_schema(dc(cls), base_schema)
 
 
-def add_schema(clazz: type) -> type:
+def add_schema(
+        cls: Type[_U] = None,
```

Comment:
Or maybe

```suggestion
        cls: _U = None,
```

Then `) -> _U` below and

```python
_U = TypeVar('_U', type)
```

above? Not a big deal though.

Reply:
Having `_U = TypeVar('_U')` allows us to reuse the same type variable in different contexts.

Reply:
Sorry, edited my suggestion to use a separate type variable (`_C`). Up to you.

Reply:
I think we can just keep it this way.